src/native/libs/System.Security.Cryptography.Native/pal_evp_mac.c

```diff
@@ -0,0 +1,383 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "openssl.h"
+#include "pal_evp_mac.h"
+#include "pal_utilities.h"
+
+#include <assert.h>
+
+EVP_MAC* CryptoNative_EvpMacFetch(const char* algorithm, int32_t* haveFeature)
+{
+    assert(haveFeature);
+
+#ifdef NEED_OPENSSL_3_0
+    if (API_EXISTS(EVP_MAC_free))
```

Comment:
While it's probably true that free and fetch came at the same time, this should probably say fetch.

```suggestion
    if (API_EXISTS(EVP_MAC_fetch))
```